documentation/modules/ROOT/pages/connectors/vitess.adoc

```diff
@@ -31,15 +31,34 @@ The connector gives you the flexibility to choose to subscribe to the `MASTER` n
+The underlying MySQL normally purges binary logs after some period of time. This means that the connector does not have the complete history of all changes that have been made to the database. Therefore, when the connector first connects to a particular database, it starts by performing a consistent snapshot of each of the database. After the connector completes the snapshot, it continues streaming changes from the exact point at which the snapshot was made. This way, the connector starts with a consistent view of all of the data, and does not omit any changes that were made while the snapshot was being taken.
+
+The connector is tolerant of failures. As the connector reads changes and produces events, it records the VGTID position for each event. If the connector stops for any reason (including communication failures, network problems, or crashes), upon restart the connector continues reading the WAL where it last left off. This does not include snapshots. If the connector stops during a snapshot, upon restart the connector does not continue performing snapshots where it last left off.
+We'll talk later about how the connector behaves xref:{link-vitess-connector}#vitess-when-things-go-wrong[when things go wrong].
 
 // Type: assembly
 // ModuleID: how-debezium-vitess-connectors-work
 // Title: How {prodname} Vitess connectors work
 [[how-the-vitess-connector-works]]
 == How the connector works
 
-To optimally configure and run a {prodname} Vitess connector, it is helpful to understand how the connector streams change events, determines Kafka topic names, and uses metadata.
+To optimally configure and run a {prodname} Vitess connector, it is helpful to understand how the connector performs snapshots, streams change events, determines Kafka topic names, and uses metadata.
+
+// Type: concept
+// ModuleID: how-debezium-vitess-connectors-perform-snapshot
+// Title: How {prodname} Vitess connectors perform a snapshot
+[[vitess-snapshot]]
+=== Snapshot
+
+Most MySQL servers are configured to not retain the complete history of the database in the binary logs. This means that the connector would be unable to see the entire history of the database by reading only the binary logs. Consequently, the first time that the connector starts, it performs an initial consistent snapshot of the database. You can change this behavior by setting the xref:{link-vitess-connector}#vitess-property-snapshot-mode[`snapshot.mode` connector configuration property] to a value other than `initial`.
+This snapshot feature is built on link:https://vitess.io/docs/design-docs/vreplication/vstream/vscopy/[VStream Copy] introduced in version 7.0.
+
+[NOTE]
+====
+In Vitess 15 or earlier, it is not possible to resume `VStream Copy.`
+Starting in Vitess 16, it will support resuming this initial copy phase if it's interrupted for any reason.
```

Comment:
```suggestion
Beginning with Vitess 16, you can resume the initial copy phase if it's interrupted for any reason.
```

It's not clear from the original text, but with Vitess 16 and later, is the initial copy phase resumed automatically following an interruption?

Reply:
@yoheimuta Could you please comment on that question? Thanks a lot.

Reply:
Thank you for your reviews!

> with Vitess 16 and later, is the initial copy phase resumed automatically following an interruption?

@roldanbob No. In order to achieve this, the connector will also need to be updated.

Reply:
> Thank you for your reviews!

You're very welcome!

> > with Vitess 16 and later, is the initial copy phase resumed automatically following an interruption?
> 
> @roldanbob No. In order to achieve this, the connector will also need to be updated.

I'm not sure that I follow. Do you mean that:

1. You must restart the connector to resume a snapshot that was interrupted?
2. You must update the Debezium Vitess connector to the version that this change is applied to (2.1?)?
3.  Or is it actually both?: That is, if you're running earlier versions of the connector against Vitess 16 or later, you cannot resume a snapshot that was interrupted. You must be running 2.1 or later _and_ restart the connector to resume an interrupted snapshot.

Reply:
It means the release of Vitess 16 is a blocker for me to implement this feature fault torelantly.